chalk-solve/src/clauses.rs

```diff
@@ -283,7 +284,25 @@ fn program_clauses_that_could_match<I: Interner>(
                     ..
                 })
                 | TyData::Alias(AliasTy::Opaque(OpaqueTy { opaque_ty_id, .. })) => {
-                    db.opaque_ty_data(*opaque_ty_id).to_program_clauses(builder);
+                    db.opaque_ty_data(*opaque_ty_id)
+                        .to_program_clauses(builder, environment);
+                }
+                _ => {}
+            }
+
+            // We don't actually do anything here, but we need to record the types it when logging
```

Comment:
This surprised me a bit. But I can't think of a better way to handle it.
The problem here is that `foo: Sized` (where `foo` is a function) doesn't actually call `fn_def_datum`. So the `LoggingRustIrDatabase` doesn't pick it up. This could be removed when the that actually records the *goals* too.